en/debug/porting-guide.md

```diff
@@ -1,25 +1,51 @@
-# Porting Guide
+# Flight Controller Porting Guide
 
-PX4 consists of two main layers: The board support and middleware layer on top of the host OS \(NuttX, Linux or any other POSIX platform like Mac OS\). And the applications \(in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).
+This topic for developers who want to port PX4 to work with *new* flight controller hardware.
+
+## Architecture
+
+PX4 consists of two main layers: The board support and middleware layer on top of the host OS (NuttX, Linux or any other POSIX platform like Mac OS). And the applications (in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).
 
 This guide is focused only on the middleware as the applications will run on any board target.
 
 ## NuttX Boards
 
-The main files for a NuttX board are located at \(for the FMUv5 example\):
+The location of the main files for a NuttX board are:
+
+* Board startup and PX4 board configuration: [src/drivers/boards](https://github.com/PX4/Firmware/tree/master/src/drivers/boards). It contains bus and GPIO mappings and the board initialization code. 
+  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5). 
+* NuttX board config: [nuttx-configs](https://github.com/PX4/Firmware/tree/master/nuttx-configs). The OS gets loaded as part of the application build. 
+  * FMUv5 example: [nuttx-configs/px4fmu-v5/include/board.h](https://github.com/PX4/Firmware/blob/master/nuttx-configs/px4fmu-v5/include/board.h).
+* NuttX OS config (created with menuconfig): [nuttx-configs/px4fmu-v5/nsh/defconfig](https://github.com/PX4/Firmware/blob/master/nuttx-configs/px4fmu-v5/nsh/defconfig).
+* Linker scripts and other required settings: [nuttx-configs](https://github.com/PX4/Firmware/tree/master/nuttx-configs). 
+  * FMUv5 example: [nuttx-configs/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/nuttx-configs/px4fmu-v5).
+* Boot file system (startup script): [ROMFS/px4fmu\_common](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common)
+* Board build configuration: [cmake/configs/nuttx\_px4fmu-v5\_default.cmake](https://github.com/PX4/Firmware/blob/master/cmake/configs/nuttx_px4fmu-v5_default.cmake).
+* Drivers: [src/drivers](https://github.com/PX4/Firmware/tree/master/src/drivers).
+* Reference config: Running `make px4fmu-v4_default` builds the FMUv4 config, which is the current NuttX reference configuration.
```

Comment:
Is FMUv4 still the NuttX reference. If not, I could upgrade this to FMUv5

Reply:
Either should be fine.